src/impl/conv_direct.jl

```diff
@@ -81,6 +81,11 @@ function conv_direct!(
     # Use `calc_padding_regions` to determine where we do or don't need to worry about padding
     padded_regions, central_region = calc_padding_regions(cdims)
 
+    # Set outputs to zero to support custom datatypes (https://github.com/FluxML/NNlib.jl/issues/490)
+    if iszero(beta)
```

Comment:
If this zeroing impacts performance, perhaps we could check here whether `eltype(y) <: AbstractFloat` and avoid it then?

Reply:
I could also have subtyped `MyFloat` from the test set as an `AbstractFloat`, so that's not enough. What you would actually want to test for is whether the elements are `#undef` or not.

Reply:
Not enough but better than nothing. The right test is probably `eltype(y) <: Union{...}` and we need to figure out all the types for which we can multiply an `undef` element by zero and get away with it.

Reply:
> Not enough but better than nothing. 

To my understanding, `conv_direct!` is supposed to be a fallback implementation for odd datatypes that is implemented naively using for-loops:

> ```Julia
> """
>     conv_direct!(y, x, w, cdims; alpha=1, beta=0)
>
> Direct convolution implementation; used for debugging, tests, and mixing/matching of 
> strange datatypes within a single convolution. Uses naive nested for loop implementation

so I would argue that a large gain in compatibility at the cost of a most likely negligible decrease in performance is perfectly acceptable.